### PR TITLE
Sign Up: Plans step for domain upgrades

### PR DESCRIPTION
**Pull request by rodrigoi**
Adding a Domain Registration or Domain Mapping, the signup flows always `goToNextStep` to show the plans page.

Plans and Plans compare on the signup flows hide the Free plan if the user has a domain upgrade.

Test live: https://calypso.live/?branch=add/signup-plans-step

### CODE REVIEW

**Comment by apeatling** [comment]
Confirmed that I saw the plans step with Premium and Business after selecting a .com domain name. When I got to the cart I correctly saw the business plan, although did not go through with the purchase.

**Comment by umurkontaci** [comment]
To clarify, is the goal of the PR to remove the auto-bundling plan with a domain and hide free option on the plans page?

**Comment by roundhill** [comment]
I tested both premium and business flows and both were added to the cart successfully. 

Code changes looks good to me, although I haven't worked with the domains code before.

**Comment by roundhill** [comment]
@umurkontaci We want the user to choose between premium and business instead of automatically being assigned to premium after the domain step.

File: client/signup/steps/plans/index.jsx [hunk 1]
```diff
@@ -81,13 +81,21 @@ module.exports = React.createClass( {
 		analytics.tracks.recordEvent( 'calypso_signup_compare_plans_click', { location: linkLocation } );
 	},
 
+	hideFreePlan: function() {
+		if ( this.props.signupDependencies && this.props.signupDependencies.domainItem ) {
+			return this.props.signupDependencies.domainItem.is_domain_registration;
+		}
+
+		return this.props.hideFreePlan;
+	},
```
Comment (umurkontaci) [inline client/signup/steps/plans/index.jsx]: Overwriting props depending on something else makes things quite complicated very fast. Is it possible to move this to a prop coming from the parent component?

**Comment by apeatling** [comment]
This looks good to me, I haven't run into any issues testing and trying to break it. After @umurkontaci's request is done, I think this is good to merge. :+1:

**Comment by umurkontaci** [comment]
Testing ✅ 
After the code change related to props it's good to 🚢

**Comment by apeatling** [comment]
💥

**Pull request merged**